Add arm64 support to docker image creating tasks

mari-crl opened the pull request:
This changes the x86_64 image in nightly and final build steps to be prefixed
with "amd64-", and adds an arm64 image which is prefixed with "arm64-".
The unprefixed image, which was previously the x86_64 image, is now a
multi-architecture image which points to the two prefixed images.

This change also splits build_docker_image into build_docker_image_arm64
and build_docker_image_x86_64. The Build and Test Docker TeamCity builds
have been split accordingly, into arm64 and x86_64 variants for each.
These builds do not build/test the combined image, only the individual
platforms' images.

Part of #60778 

Release note (general change): Cockroach Docker images are now multi
architecture manifests supporting the x86_64 (amd64) and arm64
architectures.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/89548)
<!-- Reviewable:end -->

rickystewart commented on build/teamcity/cockroach/ci/builds/build_docker_image_arm64.sh:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+source "$(dirname "${0}")/build_docker_image.sh" arm64
```
nit: newline at end of file

mari-crl replied:
Done!

mari-crl commented on build/teamcity/cockroach/ci/builds/build_docker_image_x86_64.sh:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+source "$(dirname "${0}")/build_docker_image.sh" amd64
```
And done!

jlinder reviewed (changes requested):
This is great! I've a couple commit message requests.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89548#-:-NEGb3eN7u4R7P2XzEI-:bp4m77t)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mari-crl and @rickystewart)

___
*[`-- commits` line 3 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/89548#-NEG_85B0CbR2GSQXh1h):*
Please add more description of what the change entails. For example: that it creates a multi-arch image and that it includes the changes for publishing nightly and final images.
___
*[`-- commits` line 4 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/89548#-NEGYlM55KNrTazaD4Ah):*
Please add a `Release notes (build change):` line for this commit. See the [release notes wiki page](https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/186548364/Release+notes) for details.


<!-- Sent from Reviewable.io -->

jlinder reviewed (approved):
I'm good with this. So long as there are some passing dry runs for `Make and Publish Build` and for `Publish Cockroach Release` then this can be pushed. We can fix the `Build Docker Image (Linux arm64)` test in a subsequent PR.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89548#-:-NEbSRPpEa_AN_QRqSsL:bzaqlad)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mari-crl and @rickystewart)



<!-- Sent from Reviewable.io -->

mari-crl commented:
Your foe's weak! Get 'em, bors!

bors r+

mari-crl commented:
That's enough, Bors, come back!

bors r-

mari-crl committed:
Add arm64 support to docker image creating tasks

This changes the x86_64 image in nightly and final build steps to be prefixed
with "amd64-", and adds an arm64 image which is prefixed with "arm64-".
The unprefixed image, which was previously the x86_64 image, is now a
multi-architecture image which points to the two prefixed images.

This change also splits build_docker_image into build_docker_image_arm64
and build_docker_image_x86_64. The Build and Test Docker TeamCity builds
have been split accordingly, into arm64 and x86_64 variants for each.
These builds do not build/test the combined image, only the individual
platforms' images.

Part of CRDB-3143

Release note (general change): Cockroach Docker images are now multi
architecture manifests supporting the x86_64 (amd64) and arm64
architectures.

jlinder reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89548#-:-NEndRhcD9jI246tDjBc:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @mari-crl and @rickystewart)



<!-- Sent from Reviewable.io -->

rail commented:
bors r=jlinder,rickystewart
